firstvoices/backend/tests/test_apis/test_site_features_api.py

```diff
@@ -0,0 +1,159 @@
+import pytest
+
+from backend.models.constants import AppRole, Visibility
+from backend.models.sites import SiteFeature
+from backend.tests import factories
+
+from .base_api_test import BaseUncontrolledSiteContentApiTest
+
+
+class TestSiteFeatureEndpoints(BaseUncontrolledSiteContentApiTest):
+    """
+    End-to-end tests that check the site feature endpoint for expected behavior.
+    """
+
+    API_LIST_VIEW = "api:sitefeature-list"
+    API_DETAIL_VIEW = "api:sitefeature-detail"
+    TEST_KEY = "test"
+
+    model = SiteFeature
+
+    def get_lookup_key(self, instance):
+        return instance.key
+
+    def create_minimal_instance(self, site, visibility):
+        return factories.SiteFeatureFactory.create(site=site)
+
+    def get_expected_detail_response(self, instance, site):
+        return {
+            "created": instance.created.astimezone().isoformat(),
+            "createdBy": instance.created_by.email,
+            "lastModified": instance.last_modified.astimezone().isoformat(),
+            "lastModifiedBy": instance.last_modified_by.email,
+            "id": str(instance.id),
+            "url": f"http://testserver{self.get_detail_endpoint(instance.key, instance.site.slug)}",
+            "site": {
+                "id": str(site.id),
+                "url": f"http://testserver/api/1.0/sites/{site.slug}",
+                "title": site.title,
+                "slug": site.slug,
+                "visibility": instance.site.get_visibility_display().lower(),
+                "language": site.language.title,
+            },
+            "key": instance.key,
+            "isEnabled": instance.is_enabled,
+        }
+
+    def get_expected_response(self, instance, site):
+        return self.get_expected_detail_response(instance, site)
+
+    def get_valid_data(self, site=None):
+        return {
+            "key": self.TEST_KEY,
+            "isEnabled": True,
+        }
+
+    def get_valid_data_with_nulls(self, site=None):
```

Comment:
Rather than having this fake test, please just override and skip the null value tests